receiver/rabbitmqreceiver/config.go

```diff
@@ -31,20 +31,25 @@ type Config struct {
 	confighttp.ClientConfig        `mapstructure:",squash"`
 	Username                       string              `mapstructure:"username"`
 	Password                       configopaque.String `mapstructure:"password"`
+	EnableNodeMetrics              bool                `mapstructure:"enable_node_metrics"` // Added flag for node metrics
 	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
 }
 
 // Validate validates the configuration by checking for missing or invalid fields
 func (cfg *Config) Validate() error {
 	var err []error
+
+	// Validate username
```

Comment:
Please move those changes to a separate PR, same for both comment additions down below.